include/ensmallen_bits/agemoea/agemoea_impl.hpp

```diff
@@ -0,0 +1,806 @@
+      double value, perturbationFactor;
+      if (rand < 0.5)
+      {
+        value = 2.0 * rand + (1.0 - 2.0 * rand) *
+            std::pow(upperDelta, distributionIndex + 1.0);
+        perturbationFactor = std::pow(value, mutationPower) - 1.0;
+      }
+      else
+      {
+        value = 2.0 * (1.0 - rand) + 2.0 *(rand - 0.5) *
+            std::pow(lowerDelta, distributionIndex + 1.0);
+        perturbationFactor = 1.0 - std::pow(value, mutationPower);
+      }
+
+      candidate(geneIdx) += perturbationFactor * geneRange;
+    }
+    //! Enforce bounds.
+    candidate = arma::min(arma::max(candidate, lowerBound), upperBound);
+}
+
+template <typename MatType>
+inline void AGEMOEA::NormalizeFront(
+      std::vector<arma::Col<typename MatType::elem_type> >& calculatedObjectives,
+      arma::Col<typename MatType::elem_type>& normalization,
+      const std::vector<size_t>& front,
+      const arma::Row<size_t>& extreme)
+{
+  arma::Mat<typename MatType::elem_type> vectorizedObjectives(numObjectives, 
+      front.size());
+  for (size_t i = 0; i < front.size(); i++)
+  {
+    vectorizedObjectives.col(i) = calculatedObjectives[front[i]];
+  }
+
+  if(front.size() < numObjectives)
+  {
+    normalization = arma::max(vectorizedObjectives, 1);
+    return;
+  }
+  arma::Col<typename MatType::elem_type> temp;
+  arma::uvec unique = arma::find_unique(extreme);
+  if (extreme.n_elem != unique.n_elem)
+  {
+    normalization = arma::max(vectorizedObjectives, 1);
+    return;
+  }
+  arma::Col<typename MatType::elem_type> one(front.size(), arma::fill::ones);
+  arma::Col<typename MatType::elem_type> hyperplane = arma::solve(
+    vectorizedObjectives.t(), one);
+  if (hyperplane.has_inf() || hyperplane.has_nan() || (arma::accu(hyperplane < 0.0) > 0))
+  {
+    normalization = arma::max(vectorizedObjectives, 1);
+  }
+  else
+  {
+    normalization = 1. / hyperplane;   
+    if (hyperplane.has_inf() || hyperplane.has_nan())
+    {    
+      normalization = arma::max(vectorizedObjectives, 1);
+    }
+  }
+  normalization = normalization + (normalization == 0);
+}
+
+template <typename MatType>
+inline double AGEMOEA::GetGeometry(
+    std::vector<arma::Col<typename MatType::elem_type> >& calculatedObjectives,
+    const std::vector<size_t>& front,
+    const arma::Row<size_t>& extreme)
+{
+  arma::Row<typename MatType::elem_type> d;
+  arma::Col<typename MatType::elem_type> zero(numObjectives, arma::fill::zeros);
+  arma::Col<typename MatType::elem_type> one(numObjectives, arma::fill::ones);
+
+  PointToLineDistance<MatType> (d, calculatedObjectives, front, zero, one);
+
+  for (size_t i = 0; i < extreme.size(); i++)
+  {
+    d[extreme[i]] = arma::datum::inf;
+  }
+  size_t index = arma::index_min(d);
+  double avg = arma::accu(calculatedObjectives[front[index]]) / static_cast<double> (numObjectives); 
+  double p = std::log(numObjectives) / std::log(1.0 / avg);
+  if (p <= 0.1 || std::isnan(p)) {p = 1.0;}
```

Comment:
```suggestion
  if (p <= 0.1 || std::isnan(p))
    p = 1.0;
```
Seems clearer to just wrap the line to me.